testdata/go/v4-alpha/monitoring/memcached-operator/controllers/memcached_controller.go

```diff
@@ -83,11 +88,23 @@ type MemcachedReconciler struct {
 func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 
+	// Check if prometheus rule already exists, if not create a new one
```

Comment:
not only you should check if the rule exists, but also if it was externally changed, and if so revert it to the initial configuration

Reply:
I added it, please let me know the change makes sense.

Reply:
looks good